fendermint/vm/interpreter/src/bytes.rs

```diff
@@ -125,13 +126,17 @@ where
     }
 
     /// Parse messages in the block, reject if unknown format. Pass the rest to the inner `ChainMessage` interpreter.
-    async fn process(&self, state: Self::State, msgs: Vec<Self::Message>) -> anyhow::Result<bool> {
+    async fn process(
+        &self,
+        state: Self::State,
+        msgs: Vec<Self::Message>,
+    ) -> anyhow::Result<bool, ProcessError> {
         if msgs.len() > self.max_msgs {
             tracing::warn!(
                 block_msgs = msgs.len(),
                 "rejecting block: too many messages"
             );
-            return Ok(false);
+            return Err(ProcessError::TooManyMessages(msgs.len()));
```

Comment:
This changes the return semantics of this method. In fact, this appears to be a breaking change. The caller (`App::process_proposal()`) would previously return a successful rejection to CometBFT, it now propagates the error.

I don't think the motivation (to capture a reason in the event) justifies the risk of a functional regression in this part of the codebase. Let's revert these changes and settle for not capturing the reason for the rejection in the trace.

Reply:
Yes I guess that makes sense. To me though I did not see why would we want to have it succeed in the case. Also I have changed the error handling as well. But I understand why the risk on correct migration is higher then benefit of reason.